Arcs/Arcs.cpp

```diff
@@ -933,82 +1770,132 @@ void runArcs(const std::vector<std::string>& filenames) {
-    std::vector<std::string> bamFiles = readFof(params.fofName);
-    std::copy(filenames.begin(), filenames.end(), std::back_inserter(bamFiles));
-    readBAMS(bamFiles, imap, indexMultMap, scaffSizeList, scaffSizeMap);
+        /* If barcode multiplicity file specified read it, otherwise read the reads to gather barcode multiplicity info.
+        This step is done before reading the sequences to prevent kmerization of reads with barcodes out of multiplicity range. */
+        time(&rawtime);
+        std::cout << "\n=>Preprocessing: Gathering barcode multiplicity information..." << ctime(&rawtime);
+        if(!params.multfile.empty()){
+            createIndexMultMap(params.multfile, indexMultMap);  // reading file
+        }
+        else{
+            std::cout << "Multiplicity information is being formed from reads as no barcode multiplicity file provided." << std::endl;
+            readBarcodes(filenames, indexMultMap);  // reading reads
+        }
 
-    size_t barcodeCount = countBarcodes(imap, indexMultMap);
+        /* Get the size of contigArray the prevent repetitive memory allocation. */
+        time(&rawtime);
+        std::cout << "\n=>Preprocessing: Gathering draft information..." << ctime(&rawtime) << "\n";
+        size_t size = initContigArray(params.file); 
+        contigRecord.resize(size);
 
-    if (!params.barcode_counts_name.empty()) {
+        /* Kmerization of contig ends */
+    	time(&rawtime);
+    	std::cout << "\n=>Storing Kmers from Contig ends... " << ctime(&rawtime) << std::endl;
+    	getContigKmers(params.file, kmap, contigRecord, contigToLength);
+
+        /* Kmerization of reads with barcodes in the multiplicity range. */
         time(&rawtime);
-        std::cout << "\n=> Writing reads per barcode TSV file... " << ctime(&rawtime) << "\n";
-        writeBarcodeCountsTSV(params.barcode_counts_name, indexMultMap);
+  	    std::cout << "\n=>Reading Chromium FASTQ file(s)... " << ctime(&rawtime) << std::endl;
+        readChroms(filenames, kmap, imap, indexMultMap, contigRecord);
     }
+    std::cout << "Cumulative memory usage: " << memory_usage() << std::endl;
 
+    /* Pair each scaffold according to the links(barcodes) hitting them. */
     time(&rawtime);
     std::cout << "\n=> Pairing scaffolds... " << ctime(&rawtime);
     pairContigs(imap, pmap, indexMultMap);
 
+    /* Create graph with nodes=scaffolds edges=number of links between the scaffolds.*/
     time(&rawtime);
     std::cout << "\n=> Creating the graph... " << ctime(&rawtime);
     createGraph(pmap, g);
 
+    /* If user specified, calculate distance and write to output files. */
     if (params.dist_est) {
         std::cout << "\n=> Calculating distance estimates... " << ctime(&rawtime);
-        calcDistanceEstimates(imap, indexMultMap, scaffSizeMap, g);
+        calcDistanceEstimates(imap, indexMultMap, contigToLength, g);
     }
 
-    if (!params.base_name.empty()) {
```

Comment:
Are you removing this check because you have a default value for base_name?

Reply:
Yes. At this point of program there is no case base_name can be empty. If user didn't specify it, program sets it just after reading user inputs.